backend/routes/admin.py

```
@@ -886,3 +902,843 @@ async def get_all_users(session: Dict[str, Any] = Depends(require_admin_role)) -
+                raise HTTPException(status_code=500, detail="Failed to delete category")
+            result = {"success": True, "action": "delete", "category_id": category_id}
+        else:
+            # Initialize management service for move/deactivate flows
+            from ..core.followup_management_service import FollowUpManagementService
+
+            management_service = FollowUpManagementService()
+
+            # Validate the deletion request
+            validation_result = management_service.validate_category_deletion(category_id)
+            if not validation_result.get("can_delete", False):
+                raise HTTPException(status_code=400, detail=validation_result.get("reason", "Cannot delete category"))
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)